### PR TITLE
arm64: dts: rockchip: Adopt mainline naming for Friendlyelec CM3588 NAS

### PR DESCRIPTION
**Pull request by ColorfulRhino**
Rename the dts file to match the new name from mainline kernel 6.11.
This makes it easier for the board config since no conditionals are needed.

~~Should be merged at the same time with https://github.com/armbian/build/pull/7082~~ (Due to the added compatibility patch, merging at the same time is not needed anymore)

This only applies to branch rk-6.1-rkr3 on purpose since old versions are deprecated.

### CODE REVIEW

**Review by amazingfate — APPROVED** [review]

File: arch/arm64/boot/dts/rockchip/Makefile
Comment (efectn) [inline arch/arm64/boot/dts/rockchip/Makefile:352]: I don't think this is a backward-compatibility change. What about adding fallback devicetree for the old naming?

Comment (ColorfulRhino) [reply arch/arm64/boot/dts/rockchip/Makefile:352]: @efectn  Backwards compatible in which way do you mean?

Comment (efectn) [reply arch/arm64/boot/dts/rockchip/Makefile:352]: > @efectn Backwards compatible in which way do you mean?

Systems using this devicetree name will not boot anymore after updating kernel

Comment (ColorfulRhino) [reply arch/arm64/boot/dts/rockchip/Makefile:352]: I see the problem now, thanks. This can probably be solved with a symlink of some sort.

Comment (ColorfulRhino) [reply arch/arm64/boot/dts/rockchip/Makefile:352]: I have added a link/stub dts with the old name. This _should_ work for existing installations.

**Review by amazingfate — APPROVED** [review]

**Comment by amazingfate** [comment]
LGTM, feel free to merge when https://github.com/armbian/build/pull/7082 is done.

**Review by efectn — APPROVED** [review]

**Comment by ColorfulRhino** [comment]
Let's merge this now since due to the added compatibility support, both dts names are working and thus merging at the same time with the above mentioned PR is not needed anymore.
@amazingfate You need to merge since you have write access :) I have rebased it onto the current branch.

**Review by amazingfate — APPROVED** [review]

**Pull request merged**
